### PR TITLE
Update pubsub docs

### PR DESCRIPTION
**Pull request by TheIronDev**
Update pubsub docs, resolving warnings

### CODE REVIEW

File: src/v2/providers/pubsub.ts [hunk 1]
```diff
@@ -86,29 +89,30 @@ export class Message<T> {
   }
 }
 
-/** The interface published in a Pub/Sub publish subscription. */
+/** @beta The interface published in a Pub/Sub publish subscription. */
```
Comment (colerogers) [inline src/v2/providers/pubsub.ts:92]: we should yank beta per our offline discussion

File: src/v2/providers/pubsub.ts [hunk 2]
```diff
@@ -6,7 +6,8 @@ import * as options from '../options';
 /**
  * Interface representing a Google Cloud Pub/Sub message.
  *
- * @param data Payload of a Pub/Sub message.
+ * @param data - Payload of a Pub/Sub message.
```
Comment (inlined) [inline src/v2/providers/pubsub.ts:9]: document the type param

File: src/v2/providers/pubsub.ts [hunk 3]
```diff
@@ -86,29 +89,30 @@ export class Message<T> {
...
 }
 
-/** PubSubOptions extend EventHandlerOptions but must include a topic. */
+/** @beta PubSubOptions extend EventHandlerOptions but must include a topic. */
 export interface PubSubOptions extends options.EventHandlerOptions {
   topic: string;
 }
 
-/** Handle a message being published to a Pub/Sub topic. */
+/** @beta Handle a message being published to a Pub/Sub topic. */
```
Comment (inlined) [inline src/v2/providers/pubsub.ts:103]: Here and the other overload the params and type param need to be documented

File: src/v2/providers/pubsub.ts (hunk 1 above)
Comment (TheIronDev) [reply src/v2/providers/pubsub.ts:92]: Done!

File: src/v2/providers/pubsub.ts (hunk 2 above)
Comment (TheIronDev) [reply src/v2/providers/pubsub.ts:9]: Updated... but there is likely a much better definition.

File: src/v2/providers/pubsub.ts (hunk 3 above)
Comment (TheIronDev) [reply src/v2/providers/pubsub.ts]: Done!

File: src/v2/providers/pubsub.ts [hunk 4]
```diff
@@ -3,10 +3,27 @@ import { ManifestEndpoint } from '../../runtime/manifest';
 import { CloudEvent, CloudFunction } from '../core';
 import * as options from '../options';
 
+/**
+ * A PubSub Topic is:
+ * <ul>
+ * A resource that you can publish messages to and then consume those messages via subscriptions.
+ * An isolated data stream for pubsub messages.
```
Comment (egilmorez) [inline src/v2/providers/pubsub.ts:10]: Suggest Pub/Sub here, unless we mean a backticked literal.

File: src/v2/providers/pubsub.ts [hunk 5]
```diff
@@ -86,29 +103,51 @@ export class Message<T> {
...
+  topic: PubSubTopic,
   handler: (event: CloudEvent<MessagePublishedData<T>>) => any | Promise<any>
 ): CloudFunction<CloudEvent<MessagePublishedData<T>>>;
 
-/** Handle a message being published to a Pub/Sub topic. */
+/**
+ * Handle a message being published to a Pub/Sub topic.
+ * @param options - Option containing information (topic) for event
+ * @param handler - runs every time a Cloud Pub/Sub message is published
+ * @param T - Type representing `Message.data`'s json format
```
Comment (egilmorez) [inline src/v2/providers/pubsub.ts:138]: Suggest "JSON" or `json` if we want the literal.

File: src/v2/providers/pubsub.ts [hunk 6]
```diff
@@ -86,29 +103,51 @@ export class Message<T> {
...
+  /** The Pub/Sub topic to watch for message events */
+  topic: PubSubTopic;
 }
 
-/** Handle a message being published to a Pub/Sub topic. */
+/**
+ * Handle a message being published to a Pub/Sub topic.
+ * @param topic - The Pub/Sub topic to watch for message events.
+ * @param handler - runs every time a Cloud Pub/Sub message is published
+ * @param T - Type representing `Message.data`'s json format
```
Comment (egilmorez) [inline src/v2/providers/pubsub.ts:127]: Suggest "JSON" or `json` if we want the literal.

File: src/v2/providers/pubsub.ts [hunk 7]
```diff
@@ -86,29 +103,51 @@ export class Message<T> {
   }
 }
 
-/** The interface published in a Pub/Sub publish subscription. */
+/**
+ * The interface published in a Pub/Sub publish subscription.
+ * @param T - Type representing `Message.data`'s json format
```
Comment (egilmorez) [inline src/v2/providers/pubsub.ts:108]: Suggest "JSON" or `json` if we want the literal.

File: src/v2/providers/pubsub.ts [hunk 8]
```diff
@@ -3,10 +3,27 @@ import { ManifestEndpoint } from '../../runtime/manifest';
...
+ * Resource that listens to the messages published by exactly one topic.
+ */
+export type PubSubSubscription = string;
+
 /**
  * Interface representing a Google Cloud Pub/Sub message.
  *
- * @param data Payload of a Pub/Sub message.
+ * @param data - Payload of a Pub/Sub message.
+ * @param T - Type representing `Message.data`'s json format
```
Comment (egilmorez) [inline src/v2/providers/pubsub.ts:26]: Suggest "JSON" or `json` if we want the literal.

**Review by egilmorez — COMMENTED** [review]
Some thoughts, thanks!

File: src/v2/providers/pubsub.ts (hunk 5 above)
Comment (TheIronDev) [reply src/v2/providers/pubsub.ts:138]: Picked JSON (did not see other occurrences of `json`)

File: src/v2/providers/pubsub.ts (hunk 6 above)
Comment (TheIronDev) [reply src/v2/providers/pubsub.ts:127]: JSON!

File: src/v2/providers/pubsub.ts (hunk 7 above)
Comment (TheIronDev) [reply src/v2/providers/pubsub.ts:108]: JSON~! 🎉

File: src/v2/providers/pubsub.ts (hunk 8 above)
Comment (TheIronDev) [reply src/v2/providers/pubsub.ts:26]: JSON!! 🎉 🎉

**Comment by TheIronDev** [comment]
> Some thoughts, thanks!

Thank you!!!

**Review by egilmorez — APPROVED** [review]
LGTM, thanks!

**Pull request merged**